apps/firebase/rules.bolt

```diff
@@ -0,0 +1,224 @@
+///////////////////////////////////////////////////////////////////////////
+// HELPER FUNCTIONS
+///////////////////////////////////////////////////////////////////////////
+
+// auth helpers
+
+isLoggedIn() { auth.uid !== null }
+
+// general helpers
+
+orZero(value) { valueOr(value, 0) }
+valueOr(value, defaultValue) { value == null ? defaultValue : value }
+isUnchanged(value) { value == prior(value) }
+isIncreased(value) { orZero(value) > orZero(prior(value)) }
+isIncreasedBy(value, increment) { orZero(value) >= orZero(prior(value)) + increment}
+isNotRemoved(value) { value != null || prior(value) == null }
+
+// Row count helpers
+
+getRecordAsInt(record) { record == null ? 0 : 1 }
+sumRowCounts(rowCount) { 
+  orZero(rowCount['0']) + orZero(rowCount['1']) + 
```

Comment:
I don't know much about bolt, Is there any way in the bolt grammar to write this more concisely so that it is easier to change the number of shards if we need to?